File: nvpl_blas/f77/scasum.f
!******************************************************************************
! Content:
!     This example demonstrates use of API as below:
!     SCASUM 
!
!******************************************************************************
      program   SCASUM_MAIN
*
      implicit none

      include "nvpl_blas_f77_blas.fi"

*       Intrinsic Functions
      intrinsic        abs

      integer          n, incx, i
      parameter        (n=5, incx=1)
      real             result
      integer          xsize
      parameter        (xsize=1+(n-1)*abs(incx))
      complex          x(xsize)
      data             (x(i), i=1,xsize)/1.0,-3.7,4.3,0.0,5.0/

*       External Subroutines
      external         print_cvector 
*
*      Executable Statements
*
      print*
      print 99
*
*       Print input data
      print*
      print 100, n, incx
      call print_cvector(x,n,incx,'X')
*
*      Call SCASUM subroutine
      result = SCASUM(n,x,incx)
*
      print*
      print 101, result 

      stop
 99   format('Example: SCASUM for computing the sum of the absolute'
     &       ' values of the elements of vector')
 100  format('#### args: n=',i1,', incx=',i1)
 101  format('The sum of the absolute values of the elements of'
     &       ' vector: ', f6.2)
      end

File: nvpl_blas/c/scasum.c
/******************************************************************************
 * Content: 
 *     This example demonstrates use of API as below:
 *     cblas_scasum
 *
 ******************************************************************************/
#include "example_helper.h"

int main() {
    float result;
    nvpl_int_t N = 5;
    nvpl_scomplex_t * X;
    nvpl_int_t incX = 1;

    printf("\nExample: cblas_scasum for computing the sum of the absolute values of the elements of vector\n\n");
    printf("#### args: n=%" PRId64 ", incx=%" PRId64 "\n", (int64_t)N, (int64_t)incX);

    nvpl_int_t len_x = 1 + (N - 1) * labs(incX);
    // allocate memory
    X = (nvpl_scomplex_t *)malloc(len_x * sizeof(nvpl_scomplex_t));

    // fill data
    fill_cvector(X, N, incX);

    // print input data
    print_cvector(X, N, incX, "X");

    // call cblas_scasum
    result = cblas_scasum(N, X, incX);

    // print result
    printf("\nThe sum of the absolute values of the elements of vector: %f\n", result);

    // release memory
    free(X);
    return 0;
}
